test/es6/expo.js

```diff
@@ -34,18 +34,52 @@ var tests = [
             var a = 2;
             a**= 4;
             assert.areEqual(Math.pow(2,4), a, "Exponentiation assignment works as intended");
+        
             var b = -2;
             b **= -4;
             assert.areEqual(Math.pow(-2,-4), b,"Exponentiation assignment works as intended");
+
+            var c = -3;
+            var d = 0;
+            d = ++c**3;
+            assert.areEqual(Math.pow(-2,3), d,"Exponentiation assignment works as intended with preincrement operator");
+  
+            c = -3;
+            d = --c**3;
+            assert.areEqual(Math.pow(-4,3), d,"Exponentiation assignment works as intended with predecrement operator");
+
+            c = -3;
+            d = c++**3;
+            assert.areEqual(Math.pow(-3,3), d,"Exponentiation assignment works as intended with postincrement operator");
+
+            c = -3;
+            d = 2**++c;
+            assert.areEqual(Math.pow(2,-2), d,"Exponentiation assignment works as intended with preincrement operator on the right hand side");
+
+            c = -3;
+            d = 2**c++;
+            assert.areEqual(Math.pow(2,-3), d,"Exponentiation assignment works as intended with postincrement operator on  the right hand side");
+            
         }
     },
     {
         name: "Exponentiation operator must be evaluated before multiplication ",
         body: function () {
-            assert.areEqual((2 * 3) ** 2, 2 * 3 ** 2, "Exponentiation operator has lower precedence than multiplicative operator *");
+            assert.areEqual(2 * (3 ** 4), 2 * 3 ** 4, "Exponentiation operator has higher precedence than multiplicative operator *");
+            assert.areEqual(2 * (3 ** 4) * 5 , 2 * 3 ** 4 * 5, "Exponentiation operator has higher precedence than multiplicative operator *");            
             assert.areEqual(2 + (3 ** 2), 2 + 3 ** 2, "Exponentiation operator has higher precedence than additive operator +");
         }
     },
+    {
+        name: "Exponentiation syntax error cases ",
+        body: function () {
+            assert.throws(function () { eval("-2**3"); }, SyntaxError, "Exponentiation operator throws if the left hand side is an unary operator +", "Invalid Unary operator on the left hand side of exponentiation (**) operator");
+            assert.throws(function () { eval("+2**3"); }, SyntaxError, "Exponentiation operator throws if the left hand side is an unary operator -", "Invalid Unary operator on the left hand side of exponentiation (**) operator");

```

Comment:
These first two descriptions are reversed.  Top one says unary + when the test is -, and vice versa for the second one.